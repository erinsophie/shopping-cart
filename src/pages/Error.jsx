import { Link } from 'react-router-dom';

function Error() {
  return (
    <div className="flex flex-col h-screen p-6 gap-3">
      <p className="text-2xl">
        Oops! This page does not exist
      </p>
      <Link to="/" className="text-blue-900 font-bold ">
        <i className="fa-solid fa-arrow-left"></i> Go back to Home page
      </Link>
    </div>
  );
}

export default Error;
