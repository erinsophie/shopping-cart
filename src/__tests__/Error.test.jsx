import { describe, it, expect, beforeEach } from 'vitest';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Error from '../pages/Error';

describe('Error page tests', () => {
  let container

  beforeEach(() => {
    const { container: renderedContainer } = render(
      <MemoryRouter>
        <Error />
      </MemoryRouter>
    );

    container = renderedContainer
  });

  it('Matches the snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('Displays the error message', () => {
    const errorMessage = screen.getByText('Oops! This page does not exist');
    expect(errorMessage).toBeInTheDocument();
  });

  it('Displays link to go back to the Home page', () => {
    const link = screen.getByRole('link', { name: /go back to home page/i });
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', '/');
  });
});