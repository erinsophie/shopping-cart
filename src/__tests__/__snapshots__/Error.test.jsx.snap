// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error page tests > Matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col h-screen p-6 gap-3"
  >
    <p
      class="text-2xl"
    >
      Oops! This page does not exist
    </p>
    <a
      class="text-blue-900 font-bold "
      href="/"
    >
      <i
        class="fa-solid fa-arrow-left"
      />
       Go back to Home page
    </a>
  </div>
</div>
`;
